README.md

```diff
@@ -41,23 +49,58 @@ resolved Promise.
 
 ### Generate a DID Document
 
-Generate a new DID document.
+#### Generating and Registering a Veres One DID Document
 
-* options - a set of options used when generating the DID Document
-  * didType - the type of DID to generate.
-      Options: 'nym' or 'uuid' (default: 'nym')
-  * keyType - the type of keys to generate.
-      Options: 'RsaVerificationKey2018' (default: 'RsaVerificationKey2018').
-  * passphrase - the passphrase to use to encrypt the private keys for
-      nym-based DIDs. Set to `null` if the private keys should not be encrypted.
-  * mode - the mode/environment to generate the DID in.
-      Options: 'dev', 'test', 'live' (default: 'dev').
+```js
+// Generate a new DID Document, store the private keys locally
+veresDriver
+  .generate({didType: 'nym', keyType: 'Ed25519VerificationKey2018'}) // default
+  .then(didDocument => {
+    // A new didDocument is generated. Log it to console
+    console.log('Generated:', JSON.stringify(didDocument, null, 2));
+    return didDocument;
+  })
+
+  // Now register the newly generated DID Document
+  // Use Equihash Proof of Work by default (see below)
+  .then(didDocument => veresDriver.register({ didDocument }))
+
+  // Log the results
+  .then(registrationResult => {
+    // Log the result of registering the didDoc to the VeresOne Test ledger
+    console.log('Registered!', JSON.stringify(registrationResult, null, 2));
+  })
+  .catch(console.error);
+```
+
+#### Registering a (newly generated) DID Document
+
+To register a DID Document using an Equihash proof of work:
+
+```js
+veresDriver.register({ didDocument }).then().catch();
+```
+
+To register using an Accelerator:
+
+```js
+const accelerator = 'genesis.testnet.veres.one';
+const authDoc = didDocumentFromAccelerator; // obtained previously
 
-Returns an object with:
+veresDriver.register({ didDocument, accelerator, authDoc })
+  .then(result => console.log(JSON.stringify(await result.text(), null, 2)))
```

Comment:
Can this be written to use async/await or Promises but not both?  Also, is it true that `result.text()` returns a promise?

Reply:
Please use `async/await`; they greatly improve readability/understanding.

Reply:
Heheheh, so, originally, these instructions were written as `async/await`. Except a couple of customers' developers just cut and paste the examples and tried to use await at the top level (not inside an `async` function), and got confused why it wasn't working. So I actually switched it to use .then()/.catch()

Reply:
@mattcollier 
> Also, is it true that `result.text()` returns a promise?

It does, yeah. (I double-checked with the whatwg fetch() spec, just in case.)